src/sorts/exchange/BubbleBogoSort.java

```diff
@@ -46,15 +49,12 @@ public BubbleBogoSort(ArrayVisualizer arrayVisualizer) {
     }
 
     @Override
-    public void runSort(int[] array, int sortLength, int bucketCount) throws Exception {
-        while(!this.bogoIsSorted(array, sortLength)){
-            int index1 = (int) (Math.random() * (sortLength - 1));
-            
-            Highlights.markArray(1, index1);
-            
-            if(Reads.compareValues(array[index1], array[index1 + 1]) == 1){
-                Writes.swap(array, index1, index1 + 1, 1, true, false);
-            }
+    public void runSort(int[] array, int length, int bucketCount) {
+        while (!this.isArraySorted(array, length)) {
+            int index = BogoSorting.randInt(0, length-1);
+
+            if (Reads.compareIndices(array, index, index+1, this.delay, true) > 0)
+                Writes.swap(array, index, index+1, this.delay, true, false);
```

Comment:
Currently Bubble Bogo and Exchange Bogo have comparisons be no-delay + no-mark and swaps be delay + mark. Could you try to keep it that way?

Reply:
Is there a particular reason you'd prefer such a change?

Reply:
Because I feel these sorts (Bubble Bogo and Exchange Bogo) would finish too fast with having such a small delay.

Reply:
That appears to not be the case... But the nice Bubble Bogo music is gone 😢

Reply:
oof, I will go see the difference

Reply:
It sounds noice once again 🙂 
![image](https://user-images.githubusercontent.com/42721887/116475141-0b3ec080-a83f-11eb-8eab-4230027ccded.png)

Reply:
nice